### DISCUSSION TITLE
I want to directly pass in the application object to Granian

### DISCUSSION THREAD

**Discussion by vanker-x** [emmett-framework/granian#180] [Q&A]
Hello, currently I want to use writing Python code instead of starting the service from the command line, but I found that the target parameter of the Granian class must be of the dotted string path. Can you help me check any clever way to directly pass in an application object ?
My environmental information:
OS : Macos
Python : python 3.9
granian : granian-0.7.6

Here are the effects I may want？
```python
from granian import Granian
from vank.core import Application, request, response

app = Application()


@app.route('/', methods=['GET'])
async def say_hi(*args, **kwargs):
    return response.ResponsePlain(f'Hi,your request path is {request.path}')


Granian(app).serve()  # this is what I want

```

**Comment by gi0baro** [comment]
I think #35 is a good starting point.
In essence you can use the `target_loader` and the `wrap_loader` arguments of `Granian.serve` to customise the application loading.

Beware that Granian relies on multiprocessing even when you specify a single worker, so whatever you specify in the loader needs to be pickable, and that's also the reason why the target is by default a string in Granian.

It would be nice to know the rationale behind your decision and your use-case for this, as in general relying on the CLI is easier and more portable.

**Reply by vanker-x** [reply]
I apologize for not reading the issue and I am very happy to receive your reply.
The reason why I don't want to use the command line is that debugging applications is not easy, such as the debugger for PyCharm.
